src/probnum/randprocs/markov/_transition.py

```diff
@@ -425,6 +425,20 @@ def jointly_transform_base_measure_realization_list_forward(
             out_samples.append(curr_sample)
         return out_samples
 
+    def duplicate(self, **changes):
+        # Wrap duplication into appropriate type checks.
+        duplicated = self._duplicate(**changes)
+
+        # Safeguard against inheritance of the duplication functionality
+        # which is no actual duplication hence we pretend it is not implemented.
+        if type(duplicated) is not type(self):
+            raise NotImplementedError("duplication is not implemented.")
+        return duplicated
+
+    @abc.abstractmethod
+    def _duplicate(self, **changes):
+        raise NotImplementedError
```

Comment:
the issue here is that a `duplicate()` method should return an object of the same type, so it needs to be implemented in *each* subclass (in particular, we want to avoid inheritance). This solution with an abstract method forces that at least some duplication method can be found, however, it can only be *used* if it is implemented in this particular class (all transitions implement one).

Reply:
I see the problem.

Wouldn't it be sufficient to make `duplicate` an `@abc.abstractmethod`? Then everything that subclasses `Transition` has to implement it. Or am I missing something?

Reply:
Unfortunately not, because then e.g. `LinearSDE` would be duplicated into an `SDE` if not implemented.